### PR TITLE
Converted Qt build to 3p package source.

### PR DESCRIPTION
**Pull request by cgalvan**
Added Qt build scripts for 3p package source. Only tested on Windows currently, but have converted Linux/Darwin build script as well.

- Built off of our O3DE fork of Qt 5.15.1
- Have to locally install miniconda to the temporary directory so that we have a Python 2.7 executable to build WebEngine
- On Windows, also have to download/install jom and ICU to the temporary directory
- On Windows we use the OpenSSL runtime package that we build as a dependency for linking
- On all platforms, have to copy Platform-specific cmake files to the install directory

Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

File: package-system/Qt/docker/linux-build.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+RUN update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
+
+RUN apt-get -y install wget
+RUN wget -qO - https://package.perforce.com/perforce.pubkey | apt-key add -
+RUN echo "deb http://package.perforce.com/apt/ubuntu bionic release" > /etc/apt/sources.list.d/perforce.list
+RUN apt-get update
+RUN apt-get -y install helix-p4d
+
+VOLUME /data
```
Comment (alexmontAmazon) [inline package-system/Qt/docker/linux-build.Dockerfile:25]: there's a couple of these EOF without newline, does it matter?

File: package-system/Qt/copy_platform_cmakes.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# There are some additional cmake files we need to copy per-platform to the install directory
+platform_system = platform.system().lower()
+platform_to_pal = {
+    "windows": "Windows",
+    "linux": "Linux",
+    "darwin": "Mac"
+}
+
+if not platform_system in platform_to_pal:
```
Comment (NicholasVanSickle) [inline package-system/Qt/copy_platform_cmakes.py:25]: uber nit: `if platform_system not in platform_to_pal:` is slightly more idiomatic

File: package-system/Qt/docker/linux-build.Dockerfile (hunk 1 above)
Comment (NicholasVanSickle) [reply package-system/Qt/docker/linux-build.Dockerfile:25]: I don't think it'll hurt anything, but might as well appease the diff tool and add some LFs IMO

File: package-system/Qt/docker/linux-build.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+RUN apt-get update
+
+# Qt5 build dependencies
+RUN apt-get -y install git python make g++ clang gperf bison flex libnss3-dev libxrandr-dev libdbus-1-dev libxi-dev libxtst-dev libxcomposite-dev libxcursor-dev libc++-dev libc++abi-dev libssl-dev libfontconfig1-dev libxcb1-dev libglu1-mesa-dev mesa-common-dev libxkbcommon-x11-dev libxkbcommon-dev libx11-xcb-dev libxcb-xkb-dev libxcb-icccm4-dev libxcb-keysyms1-dev libxcb-image0-dev libxcb-shm0-dev libxcb-render-util0-dev libxcb-xinerama0-dev libxcb-xinput-dev libwebp-dev libopus-dev --fix-missing
+
+# Developer Tools
+RUN apt-get -y install vim
+
+# LY build dependencies
```
Comment (NicholasVanSickle) [inline package-system/Qt/docker/linux-build.Dockerfile:11]: s/LY/O3DE/

File: package-system/Qt/docker/linux-build.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+RUN apt-get -y install clang-6.0 libc++-dev libc++abi-dev uuid-dev libz-dev libncurses5-dev libcurl4-openssl-dev libjpeg-dev libjbig-dev libpython3.7
+ 
+RUN update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+RUN update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
+
+RUN apt-get -y install wget
+RUN wget -qO - https://package.perforce.com/perforce.pubkey | apt-key add -
+RUN echo "deb http://package.perforce.com/apt/ubuntu bionic release" > /etc/apt/sources.list.d/perforce.list
+RUN apt-get update
+RUN apt-get -y install helix-p4d
```
Comment (NicholasVanSickle) [inline package-system/Qt/docker/linux-build.Dockerfile:23]: I think we can get away with no p4 now

**Review by NicholasVanSickle — APPROVED** [review]
Awesome work! This looks good to go to me, I figure we can submit this and then you can iterate on Linux/Mac if needed

File: package-system/Qt/copy_platform_cmakes.py (hunk 2 above)
Comment (cgalvan) [reply package-system/Qt/copy_platform_cmakes.py:25]: Good call, will fix!

**Comment by cgalvan** [comment]
> Was there discussion about removing the debug libs/binaries and symbols from the Qt packages to save on space? on the current rev, the symbols for the Qt5WebEngineCore alone come out to 3+GB

There have been some discussions on that, but a decision hasn't been made yet. That is out of scope for this current conversion though. This is meant to be a 1-to-1 conversion of the exact source/build scripts used to produce our latest binaries.

**Comment by cgalvan** [comment]
> trying this on mac, I get
> 
> ```
> [/Volumes/Lumberyard/o3de-repos/3p-package-source lawsonn] 14:38:50$ python3 ../3p-package-scripts/o3de_package_scripts/build_package.py --search_path . qt-5.15.2-rev5-mac
> Output folder for packages is './packages' - Override with --output_folder
> Loading darwin package lists from ....
> Loading and merging /Volumes/Lumberyard/o3de-repos/3p-package-source/package_build_list_host_darwin.json ...
> Package info file not found: /Volumes/Lumberyard/o3de-repos/3p-package-source/package-system/qt-mac/PackageInfo.json ... skipping
> ```
> 
> and thats all. did I miss something?

I hadn't updated the `package_build_list_host_darwin/linux` files since I hadn't completely tested them yet, but I will add those so you can check it out.

**Comment by lawsonamzn** [comment]
> > trying this on mac, I get
> > ```
> > [/Volumes/Lumberyard/o3de-repos/3p-package-source lawsonn] 14:38:50$ python3 ../3p-package-scripts/o3de_package_scripts/build_package.py --search_path . qt-5.15.2-rev5-mac
> > Output folder for packages is './packages' - Override with --output_folder
> > Loading darwin package lists from ....
> > Loading and merging /Volumes/Lumberyard/o3de-repos/3p-package-source/package_build_list_host_darwin.json ...
> > Package info file not found: /Volumes/Lumberyard/o3de-repos/3p-package-source/package-system/qt-mac/PackageInfo.json ... skipping
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and thats all. did I miss something?
> 
> I hadn't updated the `package_build_list_host_darwin/linux` files since I hadn't completely tested them yet, but I will add those so you can check it out.

I can help test! :)

**Review by lawsonamzn — APPROVED** [review]
tested locally by installing a version of xcode thats compatible with this chromium.
It works fine on my mac.

File: package-system/Qt/docker/linux-build.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+# Qt5 build dependencies
+RUN apt-get -y install git python make g++ clang gperf bison flex libnss3-dev libxrandr-dev libdbus-1-dev libxi-dev libxtst-dev libxcomposite-dev libxcursor-dev libc++-dev libc++abi-dev libssl-dev libfontconfig1-dev libxcb1-dev libglu1-mesa-dev mesa-common-dev libxkbcommon-x11-dev libxkbcommon-dev libx11-xcb-dev libxcb-xkb-dev libxcb-icccm4-dev libxcb-keysyms1-dev libxcb-image0-dev libxcb-shm0-dev libxcb-render-util0-dev libxcb-xinerama0-dev libxcb-xinput-dev libwebp-dev libopus-dev --fix-missing
+
+# Developer Tools
+RUN apt-get -y install vim
+
+# O3DE build dependencies
+RUN apt-get -y install build-essential
+RUN apt-get -y install libgl-dev libglu1-mesa-dev freeglut3-dev mesa-common-dev libsdl2-2.0-0 libsdl2-dev
+RUN apt-get -y install clang-6.0 libc++-dev libc++abi-dev uuid-dev libz-dev libncurses5-dev libcurl4-openssl-dev libjpeg-dev libjbig-dev libpython3.7
```
Comment (lumberyard-employee-dm) [inline package-system/Qt/docker/linux-build.Dockerfile:14]: We use `clang-12` now.

Comment (cgalvan) [reply package-system/Qt/docker/linux-build.Dockerfile:14]: Thanks will update!

**Review by lawsonamzn — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**